### PR TITLE
Support hapi auth "modes"

### PR DESCRIPTION
**Pull request by marttinen**
Added tests to check hapi's `mode: 'optional'` and `mode: 'try'`.

### CODE REVIEW

**Comment by felixheck** [comment]
@marttinen thx for the PR. can you pls fix the linting error? :)

**Comment by marttinen** [comment]
Sure, not done yet. :)

**Comment by marttinen** [comment]
Please note that on requests with invalid authentication, the result changed from 

`'Bearer strategy="keycloak-jwt", error="Invalid credentials", reason="invalid token (expired)"'`

to 

`'Bearer strategy="keycloak-jwt", reason="invalid token (expired)", error="Invalid credentials"'`

I'm not sure why the order changed, but IMO that shouldn't be an issue.

File: test/utils.spec.js [hunk 1]
```diff
@@ -27,21 +27,11 @@ test('get boom error with custom scheme', (t) => {
   }))
 })
 
-test('get boom error with default message', (t) => {
```
Comment (marttinen) [inline test/utils.spec.js:30]: Removed because duplicated test.

**Comment by felixheck** [comment]
It's because it depends if the `error` attribute is caused by the first or last argument passed to `boom.unauthorized()` :) 

so you're finished and happy with the result? ;-) 
I am.

**Comment by marttinen** [comment]
Yep, work finished and ready to be merged.

**Pull request merged**